addons/payment/models/payment_acquirer.py

```diff
@@ -45,119 +42,116 @@ class PaymentAcquirer(osv.Model):
+        help='Message displayed, if error is occur during the payment process.')
+    # Fees
+    fees_active = fields.Boolean('Add Extra Fees')
+    fees_dom_fixed = fields.Float('Fixed Domestic Fees')
+    fees_dom_var = fields.Float('Variable Domestic Fees (in percents)')
+    fees_int_fixed = fields.Float('Fixed International Fees')
+    fees_int_var = fields.Float('Variable International Fees (in percents)')
+    sequence = fields.Integer(help="Determine the display order")
+    module_id = fields.Many2one('ir.module.module', string='Corresponding Module')
+    module_state = fields.Selection('Installation State', related='module_id.state')
+    description = fields.Html('Description')
+
+    image = fields.Binary("Image", attachment=True,
         help="This field holds the image used for this provider, limited to 1024x1024px")
-    image_medium = openerp.fields.Binary("Medium-sized image", attachment=True,
-        help="Medium-sized image of this provider. It is automatically "\
-             "resized as a 128x128px image, with aspect ratio preserved. "\
+    image_medium = fields.Binary("Medium-sized image", attachment=True,
+        help="Medium-sized image of this provider. It is automatically "
+             "resized as a 128x128px image, with aspect ratio preserved. "
              "Use this field in form views or some kanban views.")
-    image_small = openerp.fields.Binary("Small-sized image", attachment=True,
-        help="Small-sized image of this provider. It is automatically "\
-             "resized as a 64x64px image, with aspect ratio preserved. "\
+    image_small = fields.Binary("Small-sized image", attachment=True,
+        help="Small-sized image of this provider. It is automatically "
+             "resized as a 64x64px image, with aspect ratio preserved. "
              "Use this field anywhere a small image is required.")
 
-    _defaults = {
-        'company_id': lambda self, cr, uid, obj, ctx=None: self.pool['res.users'].browse(cr, uid, uid).company_id.id,
-        'environment': 'test',
-        'website_published': False,
-        'auto_confirm': 'at_pay_confirm',
-        'pending_msg': '<i>Pending,</i> Your online payment has been successfully processed. But your order is not validated yet.',
-        'done_msg': '<i>Done,</i> Your online payment has been successfully processed. Thank you for your order.',
-        'cancel_msg': '<i>Cancel,</i> Your payment has been cancelled.',
-        'error_msg': "<i>Error,</i> Please be aware that an error occurred during the transaction. The order has been confirmed but won't be paid. Don't hesitate to contact us if you have any questions on the status of your order."
-    }
-
-    def _check_required_if_provider(self, cr, uid, ids, context=None):
+
+    @api.constrains('provider')
```

Comment:
So this function should check every field that has an attribute 'required_if_provider' but is triggered only if 'provider' is changed... I'm sure this is incorrect, but I'm not sure whether it is migrable in new api... @rco-odoo ?